libusb/os/darwin_usb.c

```diff
@@ -167,6 +167,60 @@ static enum libusb_error darwin_to_libusb (IOReturn result) {
   }
 }
 
+uint32_t get_running_version(void) {
+  static int32_t running_version = 0;
```

Comment:
Using a `static` here makes this not very thread safe...

I tried running the sysctlbyname + sscanf in a loop 1 million times, and it takes only about 1.3 µs on my Mac Studio.

Maybe the caching is not needed, especially if we remove the file system usage.  Is `get_running_version` called in any hot code paths?

Reply:
It helps if we have to fall back to a file even though it is not in a hot path. The main path it may be called in is the re-enumerate path which doesn't have to be fast but reading a file every time seemed like a bad idea since the answer never changes. As for the thread safety, it doesn't really matter since this is called under the first time lock so is always set by the time we are thread hot. I can change it to atomic_int32_t to make it truly safe.

Reply:
Though.... I can't remember whether we require C11 for this backend. I remember wanting to but can't remember if I ever went through with it.

Reply:
Ok, I reorganized this code to use sysctl only which removes any need to cache. I also remembered what I was doing. The first time init was calling with a lock to fill in the cached values so there was never a threading issue anyway.